File: Unit 1 Fundamentals of programming/Fundamentals Topic 3 Iteration/Topic 3 Python Programs/W3 Q3 part numbers.py
#AS AQA unit 1 W3 Q3 part numbers

oldModel = 0
total = 0
partnum = input("Enter part number: ")
while partnum != "9999":
    while len(partnum) !=4:
        partnum = input("Error: enter 4-digit number: ")
    if partnum[3] >= "6" and partnum[3]<="8":
        oldModel += 1
        print(partnum[3], oldModel)
    total += 1
    partnum = input("Enter part number: ")
#endwhile
print("Number of old models:", oldModel)
print("Total number of parts:", total)

File: A-Level/Unit 1 Fundamentals of programming/Fundamentals Topic 3 Iteration/Topic 3 VB Programs/W3 Q3 part numbers.vb
Module Module1

    Sub Main()
        'W3 Q3 part numbers

        Dim partnum As String = "xxxx"
        Dim oldModel As Integer = 0
        Dim total As Integer = 0
        Console.WriteLine(partnum & vbTab & oldModel & vbTab & total)

        'Exit condition loop.  The condition is evaluated at the end after the loop has run at least one time
        Do
            Console.WriteLine("Enter part num:")
            partnum = Console.ReadLine
            While Len(partnum) <> 4
                Console.WriteLine("Error enter 4 digit number")
                partnum = Console.ReadLine
            End While
            If Val(partnum(3)) > 5 And Val(partnum(3)) < 9 Then
                oldModel = oldModel + 1
            End If
            total = total + 1
            Console.WriteLine(partnum & vbTab & oldModel & vbTab & total)
        Loop While partnum <> 9999
        Console.WriteLine("Number of old models:" & oldModel)
        MsgBox("OK")


    End Sub

End Module
